src/org.xtuml.bp.core/models/org.xtuml.bp.core/ooaofooa/Body/Body/Body.xtuml

```diff
@@ -966,6 +966,476 @@ return false;
+     return result;
+   end if;                            
+ end for;
+ foundInstance = ::emptyInstance();
+ // check outer statements
+ select many outerStatements related by self->ACT_BLK[R650]->ACT_SMT[R602];
+ for each outerStatement in outerStatements
+   foundInstance = outerStatement.findDeclaration(
+                             declarationLineNumber: param.declarationLineNumber,
+                            declarationLineColumn: param.declarationLineColumn);
+   if(foundInstance != ::emptyInstance())
+     // there will only be one case where an instance
+     // was found
+     break;
+   end if;   
+ end for;
+ if(foundInstance != ::emptyInstance())
+   return foundInstance;
+ end if;
+ // check committed outer statements
+ select many outerStatements related by self->ACT_BLK[R666]->ACT_SMT[R602];
+ for each outerStatement in outerStatements
+   foundInstance = outerStatement.findDeclaration(
+                             declarationLineNumber: param.declarationLineNumber,
+                            declarationLineColumn: param.declarationLineColumn);
+   if(foundInstance != ::emptyInstance())
+     // there will only be one case where an instance
+     // was found
+     break;
+   end if;   
+ end for; 
+ if(foundInstance != ::emptyInstance())
+   return foundInstance;
+ end if;
+ // check committed statements
+ select many statements related by self->ACT_BLK[R601]->ACT_SMT[R602];
+ for each statement in statements
+   foundInstance = statement.findDeclaration(
+                             declarationLineNumber: param.declarationLineNumber,
+                            declarationLineColumn: param.declarationLineColumn);
+   if(foundInstance != ::emptyInstance())
+     // there will only be one case where an instance
+     // was found
+     break;
+   end if;   
+ end for; 
+ if(foundInstance != ::emptyInstance())
+   return foundInstance;
+ end if;
+ // check nested statements
+ select many statements related by self->ACT_BLK[R612]->ACT_SMT[R602];
+ for each statement in statements
+   foundInstance = statement.findDeclaration(
+                             declarationLineNumber: param.declarationLineNumber,
+                            declarationLineColumn: param.declarationLineColumn);
+   if(foundInstance != ::emptyInstance())
+     // there will only be one case where an instance
+     // was found
+     break;
+   end if;
+ end for;
+ return foundInstance;',
+	1,
+	'',
+	"50f475be-a583-4cdb-8a28-e7bebff13698",
+	0);
+INSERT INTO O_TPARM
+	VALUES ("44de3d9a-78fa-46e9-90c6-da99c4e9753f",
+	"47be481e-6e61-46a4-b087-ee4ac0755e44",
+	'declarationLineNumber',
+	"ba5eda7a-def5-0000-0000-000000000002",
+	0,
+	'',
+	"00000000-0000-0000-0000-000000000000",
+	'');
+INSERT INTO O_TPARM
+	VALUES ("46048ce5-1e6b-4fbc-82ef-b80fc0f8ce3f",
+	"47be481e-6e61-46a4-b087-ee4ac0755e44",
+	'declarationLineColumn',
+	"ba5eda7a-def5-0000-0000-000000000002",
+	0,
+	'',
+	"44de3d9a-78fa-46e9-90c6-da99c4e9753f",
+	'');
+INSERT INTO O_TFR
```

Comment:
would this operation make more sense on ACT_BLK?

Reply:
agreed, moved